lib/rules/declaration-property-value-no-unknown/index.js

```diff
@@ -108,7 +109,15 @@ const rule = (primary, secondaryOptions) => {
 				return;
 			}
 
-			const { error } = forkedLexer.matchProperty(prop, cssTreeValueNode);
+			let error;
+			const parent = decl.parent;
+
+			if (parent && parent.type === 'atrule') {
+				// @ts-expect-error -- CSSTree types are not complete
+				({ error } = forkedLexer.matchAtruleDescriptor(parent.name, prop, cssTreeValueNode));
+			} else {
+				({ error } = forkedLexer.matchProperty(prop, cssTreeValueNode));
+			}
```

Comment:
```suggestion
				const {error} = decl?.parent?.type === 'atrule'
				// @ts-expect-error -- CSSTree types are not complete
				? forkedLexer.matchAtruleDescriptor(decl.parent?.name, prop, cssTreeValueNode)
				: forkedLexer.matchProperty(prop, cssTreeValueNode);
```

I believe we can use the optional chaining operator in `15.0.0` as we dropped support for Node.js 12.

@ybiquitous may have a better suggestion for refactoring this snippet. He's amazing at them.

Reply:
We can actually omit the optional chaining in the `matchAtruleDescriptor` function call because it is already checked to exist in the ternary condition.

I refactored the code to this suggestion (minus the superfluous optional chaining operator, plus formatting via Prettier) in [`ca25f9b`](https://github.com/stylelint/stylelint/pull/6669/commits/ca25f9bb0dac9a81c3118ffe96fc20a76a00e6c0).